File: Interviews/GFG/Daily-Problems/2022/November/Day-06/Minimum-number-of-Coins.py
class Solution:
    def minPartition(self, N):
        notes, ans, i = [2000, 500, 200, 100, 50, 20, 10, 5, 2, 1], [], 0

        while N:
            j = 1
            while j <= (N // notes[i]):
                ans.append(notes[i])
                j += 1
            N %= notes[i]
            i += 1

        return ans

File: Interviews/GFG/Daily-Problems/2022/November/Day-06/Minimum-number-of-Coins.cpp
class Solution {
   public:
    vector<int> minPartition(int N) {
        vector<int> notes = {2000, 500, 200, 100, 50, 20, 10, 5, 2, 1};
        vector<int> ans;

        int i = 0;
        while (N > 0) {
            for (int j = 1; j <= (N / notes[i]); j++) {
                ans.push_back(notes[i]);
            }
            N %= notes[i++];
        }

        return ans;
    }
};